api/auditlogpersistence/v1alpha1/auditlogpersistenceconfig_types.go

```diff
@@ -0,0 +1,134 @@
+	Status AuditLogPersistenceConfigStatus `json:"status,omitempty"`
+}
+
+// AuditLogPersistenceConfigSpec defines the desired state of AuditLogPersistenceConfig
+type AuditLogPersistenceConfigSpec struct {
+	// Enabled enables or disables the audit log persistence feature globally.
+	// When disabled, no PVCs will be created and no snapshots will be taken.
+	// Defaults to false.
+	// +optional
+	Enabled bool `json:"enabled,omitempty"`
+
+	// Storage defines the PVC configuration for audit log storage.
+	// +optional
+	Storage StorageConfig `json:"storage,omitempty"`
+
+	// AuditLog defines audit log settings that will be applied to kube-apiserver.
+	// +optional
+	AuditLog AuditLogConfig `json:"auditLog,omitempty"`
+
+	// Snapshots defines snapshot configuration for crash recovery.
+	// +optional
+	Snapshots SnapshotConfig `json:"snapshots,omitempty"`
+}
+
+// StorageConfig defines PVC storage configuration
+type StorageConfig struct {
+	// StorageClassName is the name of the StorageClass to use for PVCs.
+	// If not specified, the default storage class will be used.
+	// +optional
+	StorageClassName string `json:"storageClassName,omitempty"`
+
+	// Size is the size of each PVC created for kube-apiserver pods.
+	// Must be a valid Kubernetes quantity (e.g., "5Gi", "10Gi").
+	// Defaults to "5Gi".
+	// +optional
+	Size resource.Quantity `json:"size,omitempty"`
+}
+
+// AuditLogConfig defines audit log settings
+type AuditLogConfig struct {
+	// MaxSize is the maximum size in megabytes of the audit log file before it gets rotated.
+	// This corresponds to the --audit-log-maxsize kube-apiserver argument.
+	// If not specified, defaults to 200.
+	// +kubebuilder:validation:Minimum=1
+	// +optional
+	MaxSize *int32 `json:"maxSize,omitempty"`
+
+	// MaxBackup is the maximum number of old audit log files to retain.
+	// This corresponds to the --audit-log-maxbackup kube-apiserver argument.
+	// If not specified, defaults to 10.
+	// +kubebuilder:validation:Minimum=1
+	// +optional
+	MaxBackup *int32 `json:"maxBackup,omitempty"`
+}
+
+// SnapshotConfig defines snapshot configuration
+type SnapshotConfig struct {
+	// Enabled enables or disables automatic snapshot creation on pod crashes.
+	// Defaults to false.
+	// +optional
+	Enabled bool `json:"enabled,omitempty"`
+
+	// MinInterval is the minimum time interval between snapshots for the same pod.
+	// This prevents creating too many snapshots in rapid succession.
+	// Must be a valid duration string (e.g., "1h", "30m").
+	// Defaults to "1h".
+	// +kubebuilder:validation:Pattern=`^([0-9]+(ns|us|µs|ms|s|m|h))+$`
+	// +optional
+	MinInterval string `json:"minInterval,omitempty"`
+
+	// PerPodRetentionCount is the maximum number of snapshots to retain per PVC.
+	// When this limit is reached, the oldest snapshot for that PVC will be deleted.
+	// If not specified, defaults to 10.
+	// +kubebuilder:validation:Minimum=1
+	// +optional
+	PerPodRetentionCount *int32 `json:"perPodRetentionCount,omitempty"`
```

Comment:
so validation now says `0` is not a valid value? was this intentional? I guess we don't need the pointer anymore now 😓

Reply:
Yup, it's not a valid value because otherwise, we would clean up a snapshot as soon as we create it. I think switching to a pointer is more explicit though. It might not always be clear that 0=default.